include/libsleigh.hh

```diff
@@ -33,3 +33,63 @@
 #include "types.h"
 #include "xml.hh"
 #pragma GCC diagnostic pop
+
+namespace sleigh {
+
+static const char *gSearchPaths[] = {
+    // Derived from the installation
+    SLEIGH_SPEC_INSTALL_DIR "\0",
+    // Derived from the build
+    SLEIGH_SPEC_BUILD_DIR "\0",
+    // Common install locations
+    "/usr/local/share/sleigh", "/usr/share/sleigh", "/share/sleigh"};
+
+std::string FindSpecFileInSearchPath(std::string_view file_name,
```

Comment:
Since it's just a few helpers, I just put it straight under the includes in this mega-header. If you feel that this deserves its own TU, I can make a `.cxx` somewhere containing this stuff.

Reply:
I'd prefer you put the declaration of `FindSpecFile..` in a this header, but the implementation in a `.cpp`. You can also do something like:

header:
```
extern const std::vector<std::filesystem::path> kDefaultSlaSearchPaths;

std::filesystem::path FindSpecFile(
    std::string_view file_name,
    const std::vector<std::filesystem::path> &search_paths = kDefaultSlaSearchPaths);
```

Then define the default search paths in the `.cpp` file.